### PR TITLE
Feature/red 116 exam settings

### PR DESCRIPTION
**Pull request by PanopticaRising**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "editor.wordBasedSuggestions": false
```
Comment (PanopticaRising) [inline .vscode/settings.json:2]: I could remove these, but I don't think anyone really needs this flag on.

File: src/Courses/CourseCreation/TopicCreationModal.tsx [hunk 2]
```diff
@@ -373,7 +373,7 @@ export const TopicCreationModal: React.FC<TopicCreationModalProps> = ({ unitInde
         });
         console.log(problemsWithOrdering);
```
Comment (red-tt-bear) [inline src/Courses/CourseCreation/TopicCreationModal.tsx:374]: Not in this pr but still 😈

File: src/Courses/TopicSettings/GenericFormInputs.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        type='number'
+        InputLabelProps={{style: {width: 'max-content'}}}
+    />
+);
+
+export const maxReRandomizationsField = (register: any) => (
+    <TextField 
+        name={`${examFieldNamePrefix}.maxReRandomizations`}
+        inputRef={register()}
+        label={'Max Re-Randomizations'}
```
Comment (red-tt-bear) [inline src/Courses/TopicSettings/GenericFormInputs.tsx:73]: Need to change these to version

**Review by red-tt-bear — APPROVED** [review]

File: src/Courses/CourseCreation/TopicCreationModal.tsx (hunk 2 above)
Comment (PanopticaRising) [reply src/Courses/CourseCreation/TopicCreationModal.tsx:374]: In my defense, this Modal is now deprecated!

File: src/Courses/CourseProvider.tsx [hunk 4]
```diff
@@ -55,7 +55,7 @@ export const CourseProvider: React.FC<CourseProviderProps> = ({children}) => {
             }
             setLoading(false);
         })();
-    }, [courseId]);
```
[the remark this answers is not in the document]
Comment (PanopticaRising) [reply src/Courses/CourseProvider.tsx:58]: Good catch. I had a reason for doing this, but it seems I don't any more, so I'll revert it.

File: src/Courses/TopicSettings/ExamSettings.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { Grid } from '@material-ui/core';
+import { maxGradedAttemptsPerRandomizationField, maxReRandomizationsField, randomizationDelayField, generateSwitchField, durationField } from './GenericFormInputs';
+
+interface ExamSettingsProps {
+    // This is the register function from react-hook-forms.
+    register: any;
```
[the remark this answers is not in the document]
Comment (PanopticaRising) [reply src/Courses/TopicSettings/ExamSettings.tsx:7]: We can, I had moved them around and done a bunch of cheating stuff to get this. I think passing the formState object itself is more useful than the individual functions.

There's also a hook for it, but I didn't learn about it in time to use it.

**Pull request merged**